### PR TITLE
feat(edit-ema): Enable DnD of content types to create content

### LINKED ISSUE #27001: Edit EMA: Enable Drag and Drop Content Types To Create New Contentlets
### User Story
As a user, I want to be able to drag and drop content types to add content to the page in our new page editor, so I can easily create for my page.

```[tasklist]
### Acceptance Criteria
- [x] Add a content palette from scratch ([Figma](https://www.figma.com/file/trbwjLuf5F6KaU1rFucNeg/Edit-Page?type=design&node-id=1909%3A28315&mode=design&t=BvlLRVrFSSuhbmBa-1))
- [x] User should able to drag and drop a content type to create a piece of content
```

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- Positions for containers and contentlets should be sent from the client webapp
- We need the x, y, width, height, contentlet and containers information
- We need that information in real time, when drag starts
- We need to add angular overlays with this information to be able to use it as a drop zone
- On drop we need to save the page and reload
- We have to consider slow internet and slow browsers
- Be careful with pointer events

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
- Develop content palette: 8 hrs
- Add drag and drop functionality: 12 hrs
- Send x, y, width and height: 6 hrs
- Add overlays as

### PR DESCRIPTION
**Pull request by fmontes**
### Proposed Changes
TBD

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- {{rows | json}} -->
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html:1]: Remove

File: core-web/libs/portlets/dot-ema/src/lib/shared/enums.ts [hunk 2]
```diff
@@ -3,11 +3,13 @@ export enum CUSTOMER_ACTIONS {
     ADD_CONTENTLET = 'add-contentlet', // The customer hit add button
     DELETE_CONTENTLET = 'delete-contentlet', // The customer hit delete button
     SET_URL = 'set-url', // User navigate internally within the ema
+    SET_BOUNDS = 'set-bounds', // User navigate internally within the ema
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-ema/src/lib/shared/enums.ts:6]: fix comment

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+<div *ngFor="let row of rows" [ngStyle]="getStyle(row, 'transparent')" data-type="row">
+    <div
+        *ngFor="let column of row.columns"
+        [ngStyle]="getStyle(column, 'transparent')"
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html:5]: I don't like to call methods in the template... maybe we can turn this into a pipe? @dotCMS/frontend thoughts?

Comment (zJaaal) [reply core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html:5]: This is too specific and component bounded to be a pipe in my opinion

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/edit-ema-editor.component.html [hunk 4]
```diff
@@ -44,32 +44,75 @@
...
+        </div>
+        <div
+            [attr.data-item]="{ variable: 'webPageContent', name: 'Rich Text' } | json"
+            (dragstart)="onDragStart($event)"
+            (dragend)="onDragEnd($event)"
+            data-type="content-type"
+            style="width: 80px; aspect-ratio: 1/1; background-color: chartreuse"
+            draggable="true">
+            New Rich Text
+        </div>
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/edit-ema-editor.component.html:89]: This is temp, will be replaced with the real palette component.

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html (hunk 3 above)
Comment (manuelrojas) [reply core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/ema-page-dropzone/ema-page-dropzone.component.html:4]: [ngStyle]="column | dynamicStyle:'transparent'" I like this better!

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**